integration-tests/smoke/ccip/ccip_gas_price_updates_test.go

```diff
@@ -0,0 +1,125 @@
+package smoke
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/config"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	testsetups "github.com/smartcontractkit/chainlink/integration-tests/testsetups/ccip"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// Test_CCIPGasPriceUpdates tests that chain fee price updates are propagated correctly when
+// price reaches some deviation threshold or when the price has expired.
+func Test_CCIPGasPriceUpdates(t *testing.T) {
+	lggr := logger.TestLogger(t)
+	ctx := changeset.Context(t)
+
+	var gasPriceExpiry = 5 * time.Second
+	e, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr, &changeset.TestConfigs{
+		OCRConfigOverride: func(params changeset.CCIPOCRParams) changeset.CCIPOCRParams {
+			params.CommitOffChainConfig.RemoteGasPriceBatchWriteFrequency = *config.MustNewDuration(gasPriceExpiry)
+			return params
+		},
+	})
+	state, err := changeset.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+	require.NoError(t, changeset.AddLanesForAll(e.Env, state))
+
+	allChainSelectors := maps.Keys(e.Env.Chains)
+	assert.GreaterOrEqual(t, len(allChainSelectors), 2, "test requires at least 2 chains")
+
+	sourceChain1 := allChainSelectors[0]
+	sourceChain2 := allChainSelectors[1]
+
+	feeQuoter1 := state.Chains[sourceChain1].FeeQuoter
+	feeQuoter2 := state.Chains[sourceChain2].FeeQuoter
+
+	// get initial chain fees
+	chainFee2, err := feeQuoter1.GetDestinationChainGasPrice(&bind.CallOpts{Context: ctx}, sourceChain2)
```

Comment:
nit, but it seems we use `&bind.CallOpts{Context: ctx}` multiple times, might be worth extracting to some variable or func